recipes-debian/tzdata/tzdata_debian.bbappend

```diff
@@ -0,0 +1 @@
+CONFFILES_tzdata-core += "${@ "${sysconfdir}/timezone" if bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
```

Comment:
This line doesn't work.
**timezone** will be installed even if **INSTALL_TIMEZONE_FILE** set to **"0"**

```
$: echo 'IMSTALL_TIMEZONE_FILE_pn-tzdata = "0"' >> conf/local.conf
$: bitbake -e tzdata |grep CONFFILES_tzdata-core
#     "${@ "${sysconfdir}/timezone" if bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
# $CONFFILES_tzdata-core [2 operations]
#     "${@ "${sysconfdir}/timezone" if bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
#   "${sysconfdir}/localtime ${sysconfdir}/timezone ${@ "${sysconfdir}/timezone" if bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
CONFFILES_tzdata-core="/etc/localtime /etc/timezone "
# $INSTALL_TIMEZONE_FILE [2 operations]
INSTALL_TIMEZONE_FILE="0"
# $INSTALL_TIMEZONE_FILE_pn-tzdata
INSTALL_TIMEZONE_FILE_pn-tzdata="0"
```

Reply:
Also, "${sysconfdir}}/timezone" is in FIELS_tzdata-core on meta-debian.
If INSTALL_TIMEZONE_FILE is "0" then you may need to strip it .

Reply:
Can we overwrite the meta-debian's CONFFILES in bbappend in the following manner?

```
CONFFILES_tzdata-core = "${sysconfdir}/localtime"
CONFFILES_tzdata-core += "${@ "${sysconfdir}/timezone" if bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
```

Reply:
> Can we overwrite the meta-debian's CONFFILES in bbappend in the following manner?

Alternatively, there is the following manner. (I'm fine with either.)
```
CONFFILES_tzdata-core_remove = "${@ "${sysconfdir}/timezone" if not bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
```

Reply:
Once we use the "remove", we can not override it in further bbappend. So I prefer the former one.

Reply:
I fixed this.
~~~
CONFFILES_tzdata-core += "${@ "${sysconfdir}/timezone" if bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
~~~
↓
~~~
CONFFILES_tzdata-core = "${sysconfdir}/localtime"
CONFFILES_tzdata-core += "${@ "${sysconfdir}/timezone" if bb.utils.to_boolean(d.getVar('INSTALL_TIMEZONE_FILE')) else "" }"
~~~

~~~
$ bitbake -e tzdata | grep CONFFILES_tzdata-core
# $CONFFILES_tzdata-core [3 operations]
CONFFILES_tzdata-core="/etc/localtime /etc/timezone"
~~~

Reply:
How about the result when the INSTALL_TIMEZONE_FILE is 0?

Reply:
$ echo 'INSTALL_TIMEZONE_FILE_pn-tzdata = "0"' >> conf/local.conf

$ bitbake -e tzdata | grep CONFFILES_tzdata-core
CONFFILES_tzdata-core="  /etc/localtime"